### DISCUSSION TITLE
supabase.auth.onAuthStateChange not firing when inside iframe

### DISCUSSION THREAD

**Discussion by gameveloster** [supabase/supabase#6392] [Questions]
When a Supabase-powered React app is inside an iframe, `supabase.auth.onAuthStateChange` does not fire when a user logs in. `supabase.auth.token` is also not added to local storage on log in.

The iframe pointing to the React app is injected into the webpage by a Chrome extension [`chrome-sidebar`](https://github.com/segmentio/chrome-sidebar)

```js
function App() {
  const [session, setSession] = useState(null);

  useEffect(() => {
    setSession(supabase.auth.session());
    console.log("session:", session);    // null

    supabase.auth.onAuthStateChange((_event, session) => {
      console.log("supabase.auth.onAuthStateChange");    // THIS DOES NOT RUN WHEN USER LOGS IN !!!!
      setSession(session);
    });
  }, []);

...
```

Let's say that the React app is hosted at `https://domain.com`.

When a user logs in successfully, there are 3 possible cases I've observed:

:x: **Case 1: Browser at `https://google.com`, injected iframe at `https://example.com`**
- `supabase.auth.onAuthStateChange` does not fire
- `supabase.auth.token` is not added to Local Storage

:heavy_check_mark:  **Case 2: Browser `https://example.com`, no iframe injected** 
- `supabase.auth.onAuthStateChange` fires!
- `supabase.auth.token` is added to Local Storage!

:heavy_check_mark: **Case 3: Browser and injected iframe both at `https://example.com`**
- `supabase.auth.onAuthStateChange` fires!
- `supabase.auth.token` is added to Local Storage!

How can we get `supabase.auth.onAuthStateChange` to fire when the browser is not at `https://example.com`?

Thank you!!!

**Comment by bearbricknik** [comment]
Hello, any solution to that yet?

**Comment by mandymozart** [comment]
I have the same problem. trying to run a supabase admin from inside wordpress.
